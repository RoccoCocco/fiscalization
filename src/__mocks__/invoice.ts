import { Invoice } from '../types/invoice';
import { OrderOfNotes } from '../enums/orderOfNotes';
import { PaymentTypes } from '../enums/paymentTypes';

export const mockInvoice: Required<Invoice> = {
  billNumber: {
    bussinessUnit: 'ODV1',
    number: 1,
    paymentDevice: '1',
  },
  dateTime: new Date('2014-07-15T18:00:00.000Z'),
  hasPDV: true,
  items: {
    otherTaxRate: [
      { amount: 12.1, base: 453.3, name: 'Naziv1', rate: 40.1 },
      { amount: 50.1, base: 445.1, name: 'Naziv2', rate: 27.1 },
    ],
    pdv: [
      { amount: 20.1, base: 400.1, rate: 25.1 },
      { amount: 15.44, base: 500.1, rate: 10.1 },
    ],
    pnp: [
      { amount: 10.1, base: 100.1, rate: 30.1 },
      { amount: 20.1, base: 200.1, rate: 20.1 },
    ],
  },
  marginForTaxRate: 32.0,
  noteOfOrder: OrderOfNotes.PaymentDevice,
  noteOfRedelivary: false,
  oib: '32314900695',
  oibOperative: '34562123431',
  paymentType: PaymentTypes.Cash,
  refund: [{ amount: 17.8, name: 'Refund1' }],
  taxFree: 5.1,
  taxFreeValue: 23.5,
  totalValue: 456.1,
};

export const mockInvoiceAlternate: Invoice = {
  billNumber: {
    bussinessUnit: 'ODV1',
    number: 1,
    paymentDevice: '1',
  },
  dateTime: new Date('2015-07-15T18:00:00.000Z'),
  hasPDV: false,
  items: {},
  marginForTaxRate: 32.0,
  noteOfRedelivary: true,
  oib: '32314900695',
  oibOperative: '34562123431',
  paymentType: PaymentTypes.Cash,
  taxFree: 5.1,
  taxFreeValue: 23.5,
  totalValue: 456.1,
};
