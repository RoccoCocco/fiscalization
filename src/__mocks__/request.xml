<?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
		    <soapenv:Body><tns:RacunZahtjev xmlns:tns="http://www.apis-it.hr/fin/2012/types/f73" Id="6066e77d02419">
    <tns:Zaglavlje>
        <tns:IdPoruke>0d285d65-f18a-4f60-ac5d-489f786fbc0c</tns:IdPoruke>
        <tns:DatumVrijeme>02.04.2021T11:44:29</tns:DatumVrijeme>
    </tns:Zaglavlje>
<tns:Racun>
    <tns:Oib>32314900695</tns:Oib>
    <tns:USustPdv>true</tns:USustPdv>
    <tns:DatVrijeme>15.07.2014T20:00:00</tns:DatVrijeme>
    <tns:OznSlijed>N</tns:OznSlijed>
<tns:BrRac>
    <tns:BrOznRac>1</tns:BrOznRac>
    <tns:OznPosPr>ODV1</tns:OznPosPr>
    <tns:OznNapUr>1</tns:OznNapUr>
</tns:BrRac>
    <tns:Pdv><tns:Porez>
    <tns:Stopa>25.10</tns:Stopa>
    <tns:Osnovica>400.10</tns:Osnovica>
    <tns:Iznos>20.10</tns:Iznos>
</tns:Porez>
<tns:Porez>
    <tns:Stopa>10.10</tns:Stopa>
    <tns:Osnovica>500.10</tns:Osnovica>
    <tns:Iznos>15.44</tns:Iznos>
</tns:Porez>
</tns:Pdv>
    <tns:Pnp><tns:Porez>
    <tns:Stopa>30.10</tns:Stopa>
    <tns:Osnovica>100.10</tns:Osnovica>
    <tns:Iznos>10.10</tns:Iznos>
</tns:Porez>
<tns:Porez>
    <tns:Stopa>20.10</tns:Stopa>
    <tns:Osnovica>200.10</tns:Osnovica>
    <tns:Iznos>20.10</tns:Iznos>
</tns:Porez>
</tns:Pnp>
    <tns:OstaliPor><tns:Porez>
    <tns:Naziv>Naziv1</tns:Naziv>
    <tns:Stopa>40.10</tns:Stopa>
    <tns:Osnovica>453.30</tns:Osnovica>
    <tns:Iznos>12.10</tns:Iznos>
</tns:Porez>
<tns:Porez>
    <tns:Naziv>Naziv2</tns:Naziv>
    <tns:Stopa>27.10</tns:Stopa>
    <tns:Osnovica>445.10</tns:Osnovica>
    <tns:Iznos>50.10</tns:Iznos>
</tns:Porez>
</tns:OstaliPor>
    <tns:IznosOslobPdv>23.50</tns:IznosOslobPdv>
    <tns:IznosMarza>32.00</tns:IznosMarza>
    <tns:IznosNePodlOpor>5.10</tns:IznosNePodlOpor>
    <tns:IznosUkupno>456.10</tns:IznosUkupno>
    <tns:NacinPlac>G</tns:NacinPlac>
    <tns:OibOper>34562123431</tns:OibOper>
    <tns:ZastKod>fc05f41eb2d20a6df1940b3630f5587e</tns:ZastKod>
    <tns:NakDost>false</tns:NakDost>
</tns:Racun>
<Signature xmlns="http://www.w3.org/2000/09/xmldsig#"><SignedInfo><CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/><SignatureMethod Algorithm="http://www.w3.org/2000/09/xmldsig#rsa-sha1"/><Reference URI="#6066e77d02419"><Transforms><Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/><Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/></Transforms><DigestMethod Algorithm="http://www.w3.org/2000/09/xmldsig#sha1"/><DigestValue>k4f0fqOD7J1Oou5bTV6fA2s2c4Q=</DigestValue></Reference></SignedInfo><SignatureValue>fVkZP7vS99o0BRYJOTgYnSSNZIi90n/HreoL1c0fE7JbQ+y1O7Q1e0254Vwyi3gx8H7K0iry65mRK3mLBw2/kvxLShozIHJD6nOdQYF+FwJr+FRmSnNcyclTNWsvUeoHwtlZRjs2KyJ31eKevJl+7tgCXcmlGlqNTCzfdkrGd5iDv011ME980hB7on5NAVlcxONtwUHC1W/wB/Md+eBdF5tugXXJV+sw5SIBMEVngp6PIz7ymyXtiRgrf36K2g1fvhTPYTF5HIQ9R7YR+A+cr6R1S8W7e42DGm8GXf3AA5Lu7HbKA5vEtatMJWvZnK0e8N/yCaVUhaphevapuf1F1Q==</SignatureValue><KeyInfo><X509Data><X509Certificate>
MIID9zCCAt+gAwIBAgIUGMZfw3QkXFrfXWlH1FUyknGPACYwDQYJKoZIhvcNAQEL
BQAwgYoxCzAJBgNVBAYTAkhSMREwDwYDVQQIDAhEYWxtYXRpYTEOMAwGA1UEBwwF
U3BsaXQxETAPBgNVBAoMCEF2YXJpdGlhMRQwEgYDVQQLDAtEZXZlbG9wbWVudDEQ
MA4GA1UEAwwHQVZSVEEtUjEdMBsGCSqGSIb3DQEJARYOdGVzdEB0ZXN0LnRlc3Qw
HhcNMjEwNDAyMTUzMDM5WhcNMzEwMzMxMTUzMDM5WjCBijELMAkGA1UEBhMCSFIx
ETAPBgNVBAgMCERhbG1hdGlhMQ4wDAYDVQQHDAVTcGxpdDERMA8GA1UECgwIQXZh
cml0aWExFDASBgNVBAsMC0RldmVsb3BtZW50MRAwDgYDVQQDDAdBVlJUQS1SMR0w
GwYJKoZIhvcNAQkBFg50ZXN0QHRlc3QudGVzdDCCASIwDQYJKoZIhvcNAQEBBQAD
ggEPADCCAQoCggEBALiLcdUNL68iNE+tl+/Y/kf2UYxlKMs9ZN4kkIB8V+xfmJZe
zCcJrJeho79G6dP3m3Km+e0UujEO1WoJeG9WhbMV1jcDPsH0aeeLAwYxYtyS8wC3
W+V+SlIqVPTHMw4PSQN8dSGIuGKAUfTcQNdMvd9vUOlxUqR0AxSpYBM/MA/NQsMx
XMRxCeYXTA8O5gS7XP1MR1m01PTez49J0ztIZp/1QPghXfMzWJhQ5sKoPxl9C4n7
GkkvtocFUduHaA31D0U6qQa+G98pnDx41okr+qLIS+JtYuGMGJ/i/0LN2wNcsDWK
mW8Ie6epOV2jPgpVaUkc1d84qSEpKHb+u5itvekCAwEAAaNTMFEwHQYDVR0OBBYE
FKZ8msPbaFdUkZMZHTyNNM9ufBs3MB8GA1UdIwQYMBaAFKZ8msPbaFdUkZMZHTyN
NM9ufBs3MA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEBAKijAx46
NMkS70wd9KxSg6AdpJWjU2aLlkB6CTEWEFtZUKFhzmUxrfSMp+oBiUYiXXlSDe7C
fWf7ey3xwMEzSUQW9vSKYUOHVulMVilJgTd14t7yIZQi1VxvrT2MsokYNLbXNZZq
VuO/hukbp32YwCkVqrY00Ee4Z/IWbRDQXCFKGvLwS+bx0sAG02DXTM1Mqh4CxpuV
zI4OPVXtWdmEWwth3heV6/ln95plryhMwRN9ONmk2MAJN/A7tW5WR62Og2EBtQ2x
nAkLnNGXeKD4LcT7AdY81hF+eZ4bP7uRWiAB2x8lp2D9rZxK9IH2P0tYeAddJ1nB
92j956wUMwUi0wQ=

</X509Certificate><X509IssuerSerial><X509IssuerName>OU=Development,O=Avaritia,C=HR</X509IssuerName><X509SerialNumber>18c65fc374245c5adf5d6947d4553292718f0026</X509SerialNumber></X509IssuerSerial></X509Data></KeyInfo></Signature></tns:RacunZahtjev></soapenv:Body>
		</soapenv:Envelope>
