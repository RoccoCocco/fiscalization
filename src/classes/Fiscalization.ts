import builder from 'xmlbuilder';
import uniqid from 'uniqid';

import { P12Result, getPemFromP12 } from '../libs/p12pem';
import { Invoice } from '../types/Invoice';
import { constructX509Signature } from '../utils/X509Signature';
import { constructXmlInvoiceBody } from '../utils/Body';
import { constructXmlInvoiceHeader } from '../utils/Header';

export class Fiscalization {
  private certificate: P12Result;

  public constructor(certificate: Buffer, password: string) {
    this.certificate = getPemFromP12(certificate, password);
  }

  public create(invoiceData: Invoice): string {
    const invoiceHeader = constructXmlInvoiceHeader();
    const invoiceBody = constructXmlInvoiceBody(this.certificate, invoiceData);

    const root = builder.create('soapenv:Envelope', { encoding: 'UTF-8' });

    root.attribute(
      'xmlns:soapenv',
      'http://schemas.xmlsoap.org/soap/envelope/'
    );

    const request = root.ele('soapenv:Body').ele('tns:RacunZahtjev');

    request.attribute('xmlns:tns', 'http://www.apis-it.hr/fin/2012/types/f73');
    request.attribute('Id', uniqid());

    request.importDocument(invoiceHeader);
    request.importDocument(invoiceBody);

    const invoiceXml = root.end({ pretty: true });
    const signedXml = constructX509Signature(invoiceXml, this.certificate);

    return signedXml;
  }
}
