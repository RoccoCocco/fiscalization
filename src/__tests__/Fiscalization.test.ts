import { readFileSync } from 'fs';

import { mockInvoice, mockInvoiceAlternate } from '../__mocks__/Invoice';
import { Fiscalization } from '../classes/Fiscalization';

const mockFormat = jest.fn();

jest.mock('dayjs', () => () => ({ format: mockFormat }));
jest.mock('uniqid', () => () => 'uniqid-0123456789abcdef');
jest.mock('uuid', () => ({ v4: () => 'uuid-0123456789abcdef' }));

const TEST_CERTIFICATE_PATH = `${__dirname}/../__mocks__/cert.pfx`;
const TEST_CERTIFICATE_PASS = 'apples';
const TEST_CERTIFICATE_BUFFER = readFileSync(TEST_CERTIFICATE_PATH);

describe('Fiscalization', () => {
  let fiscalization: Fiscalization;

  beforeEach(() => {
    fiscalization = new Fiscalization(
      TEST_CERTIFICATE_BUFFER,
      TEST_CERTIFICATE_PASS
    );
  });

  it('XML should match with full data', () => {
    mockFormat
      .mockReturnValueOnce('01.01.2001T01:01:01')
      .mockReturnValueOnce('02.02.2002T02:02:02');

    const xml = fiscalization.create(mockInvoice);

    expect(xml).toMatchSnapshot();
  });

  it('XML should match with partial data', () => {
    mockFormat
      .mockReturnValueOnce('01.01.2001T01:01:01')
      .mockReturnValueOnce('02.02.2002T02:02:02');

    const xml = fiscalization.create(mockInvoiceAlternate);

    expect(xml).toMatchSnapshot();
  });
});
