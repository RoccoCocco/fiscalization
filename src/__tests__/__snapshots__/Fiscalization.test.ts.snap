// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fiscalization XML should match with full data 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\">
  <soapenv:Body>
    <tns:RacunZahtjev xmlns:tns=\\"http://www.apis-it.hr/fin/2012/types/f73\\" Id=\\"uniqid-0123456789abcdef\\">
      <tns:Zaglavlje>
        <tns:DatumVrijeme>01.01.2001T01:01:01</tns:DatumVrijeme>
        <tns:IdPoruke>uuid-0123456789abcdef</tns:IdPoruke>
      </tns:Zaglavlje>
      <tns:Racun>
        <tns:BrRac>
          <tns:BrOznRac>1</tns:BrOznRac>
          <tns:OznNapUr>1</tns:OznNapUr>
          <tns:OznPosPr>ODV1</tns:OznPosPr>
        </tns:BrRac>
        <tns:DatVrijeme>02.02.2002T02:02:02</tns:DatVrijeme>
        <tns:IznosMarza>32.00</tns:IznosMarza>
        <tns:IznosNePodlOpor>5.10</tns:IznosNePodlOpor>
        <tns:IznosOslobPdv>23.50</tns:IznosOslobPdv>
        <tns:IznosUkupno>456.10</tns:IznosUkupno>
        <tns:NacinPlac>G</tns:NacinPlac>
        <tns:NakDost>false</tns:NakDost>
        <tns:Naknade>
          <tns:Naknada>
            <tns:IznosN>17.80</tns:IznosN>
            <tns:NazivN>Refund1</tns:NazivN>
          </tns:Naknada>
        </tns:Naknade>
        <tns:Oib>32314900695</tns:Oib>
        <tns:OibOper>34562123431</tns:OibOper>
        <tns:OstaliPor>
          <tns:Porez>
            <tns:Naziv>Naziv1</tns:Naziv>
            <tns:Iznos>12.10</tns:Iznos>
            <tns:Osnovica>453.30</tns:Osnovica>
            <tns:Stopa>40.10</tns:Stopa>
          </tns:Porez>
          <tns:Porez>
            <tns:Naziv>Naziv2</tns:Naziv>
            <tns:Iznos>50.10</tns:Iznos>
            <tns:Osnovica>445.10</tns:Osnovica>
            <tns:Stopa>27.10</tns:Stopa>
          </tns:Porez>
        </tns:OstaliPor>
        <tns:OznSlijed>N</tns:OznSlijed>
        <tns:Pdv>
          <tns:Porez>
            <tns:Iznos>20.10</tns:Iznos>
            <tns:Osnovica>400.10</tns:Osnovica>
            <tns:Stopa>25.10</tns:Stopa>
          </tns:Porez>
          <tns:Porez>
            <tns:Iznos>15.44</tns:Iznos>
            <tns:Osnovica>500.10</tns:Osnovica>
            <tns:Stopa>10.10</tns:Stopa>
          </tns:Porez>
        </tns:Pdv>
        <tns:Pnp>
          <tns:Porez>
            <tns:Iznos>10.10</tns:Iznos>
            <tns:Osnovica>100.10</tns:Osnovica>
            <tns:Stopa>30.10</tns:Stopa>
          </tns:Porez>
          <tns:Porez>
            <tns:Iznos>20.10</tns:Iznos>
            <tns:Osnovica>200.10</tns:Osnovica>
            <tns:Stopa>20.10</tns:Stopa>
          </tns:Porez>
        </tns:Pnp>
        <tns:USustPdv>true</tns:USustPdv>
        <tns:ZastKod>935274bbb40a30eab1255d544b4689ce</tns:ZastKod>
      </tns:Racun>
    <Signature xmlns=\\"http://www.w3.org/2000/09/xmldsig#\\"><SignedInfo><CanonicalizationMethod Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/><SignatureMethod Algorithm=\\"http://www.w3.org/2000/09/xmldsig#rsa-sha1\\"/><Reference URI=\\"#uniqid-0123456789abcdef\\"><Transforms><Transform Algorithm=\\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\\"/><Transform Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/></Transforms><DigestMethod Algorithm=\\"http://www.w3.org/2000/09/xmldsig#sha1\\"/><DigestValue>lMmoginwZSV6l3HqVq/3z4J3QJQ=</DigestValue></Reference></SignedInfo><SignatureValue>CnjFzyV6PmnB1C+4/hNETCdwxnWRoWgybC809z5k8FepE6lYjKQufTZeTvksQNI9GJkST+LCjyGLR8FBhWwrp+425qjAwm9KP6+tPHVkYDdnEI+PDceHtug4d/R8Kx0A9YQ4udg1kJHijy74g8s3fPCAfU96rI0eCnpwAJBvJLQ=</SignatureValue><KeyInfo><X509Data><X509Certificate>
MIICczCCAdygAwIBAgIEAPymiTANBgkqhkiG9w0BAQUFADBrMQswCQYDVQQGEwJV
UzEQMA4GA1UECBMHTG90aGlhbjESMBAGA1UEBxMJRWRpbmJ1cmdoMQ0wCwYDVQQK
EwRUZXN0MRgwFgYJKoZIhvcNAQkBFglmcmVkQGhvbWUxDTALBgNVBAMTBGZyZWQw
HhcNMTAwMTA4MTczNzU4WhcNMTIwMTA4MTczNzU4WjBrMQswCQYDVQQGEwJVUzEQ
MA4GA1UECBMHTG90aGlhbjESMBAGA1UEBxMJRWRpbmJ1cmdoMQ0wCwYDVQQKEwRU
ZXN0MRgwFgYJKoZIhvcNAQkBFglmcmVkQGhvbWUxDTALBgNVBAMTBGZyZWQwgZ8w
DQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBALFSdWsBxphxHYTLHM1ETlhHItfK+xsj
mcOZ9YOJHXpsONyzNyxWJZCZSK2dmwGaxqvJEsxGo/vzKvwO5K9sAT5b5xlCJmbu
JA0XKTlLC0YTi+bNLK6db0RDoIfVfMhbW4k1Gpc/vmKWDdRAGR8phwmBQc0CqX+4
K/Vj/CCMD8hpAgMBAAGjJDAiMCAGCWCGSAGG+EIBDAQTFhF3d3cuYWJ5bG9uc29m
dC5kZTANBgkqhkiG9w0BAQUFAAOBgQA4YKdsSWLah3e0Ccpdzsqrjo/QxOvlEyRy
M1n8/Dszfxik1LurcFcM8tTuyKT9aQ2EnV/id8mKx01ldzFAPkWQWU1tappOfZ/2
bHOM8gdmiyHpKCrkkvVR+YJc0x3FuzjuqJP8CkC/WdTJr3YcuLQ6K9QhPO8KIVDs
3BvXsf4F9Q==

</X509Certificate><X509IssuerSerial><X509IssuerName>OU=,O=Test,C=US</X509IssuerName><X509SerialNumber>00fca689</X509SerialNumber></X509IssuerSerial></X509Data></KeyInfo></Signature></tns:RacunZahtjev>
  </soapenv:Body>
</soapenv:Envelope>"
`;

exports[`Fiscalization XML should match with partial data 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\">
  <soapenv:Body>
    <tns:RacunZahtjev xmlns:tns=\\"http://www.apis-it.hr/fin/2012/types/f73\\" Id=\\"uniqid-0123456789abcdef\\">
      <tns:Zaglavlje>
        <tns:DatumVrijeme>01.01.2001T01:01:01</tns:DatumVrijeme>
        <tns:IdPoruke>uuid-0123456789abcdef</tns:IdPoruke>
      </tns:Zaglavlje>
      <tns:Racun>
        <tns:BrRac>
          <tns:BrOznRac>1</tns:BrOznRac>
          <tns:OznNapUr>1</tns:OznNapUr>
          <tns:OznPosPr>ODV1</tns:OznPosPr>
        </tns:BrRac>
        <tns:DatVrijeme>02.02.2002T02:02:02</tns:DatVrijeme>
        <tns:IznosMarza>32.00</tns:IznosMarza>
        <tns:IznosNePodlOpor>5.10</tns:IznosNePodlOpor>
        <tns:IznosOslobPdv>23.50</tns:IznosOslobPdv>
        <tns:IznosUkupno>456.10</tns:IznosUkupno>
        <tns:NacinPlac>G</tns:NacinPlac>
        <tns:NakDost>true</tns:NakDost>
        <tns:Naknade/>
        <tns:Oib>32314900695</tns:Oib>
        <tns:OibOper>34562123431</tns:OibOper>
        <tns:OstaliPor/>
        <tns:OznSlijed>N</tns:OznSlijed>
        <tns:Pdv/>
        <tns:Pnp/>
        <tns:USustPdv>false</tns:USustPdv>
        <tns:ZastKod>935274bbb40a30eab1255d544b4689ce</tns:ZastKod>
      </tns:Racun>
    <Signature xmlns=\\"http://www.w3.org/2000/09/xmldsig#\\"><SignedInfo><CanonicalizationMethod Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/><SignatureMethod Algorithm=\\"http://www.w3.org/2000/09/xmldsig#rsa-sha1\\"/><Reference URI=\\"#uniqid-0123456789abcdef\\"><Transforms><Transform Algorithm=\\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\\"/><Transform Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/></Transforms><DigestMethod Algorithm=\\"http://www.w3.org/2000/09/xmldsig#sha1\\"/><DigestValue>kkAhzl2XzLD+p3b1+RFy6mDThiY=</DigestValue></Reference></SignedInfo><SignatureValue>PgRLrSm10lUiOICKwveo9bkAUWT97ShHlfapCd/3GnWZjjQfn6eZO7ED/pKez0/V4E9vY9gm2u5I8RjaSqt1zwl2YK91kZdk8sFWkYUco5vZuAi1v2JPqX+fdDH6wIBk8qiJiwaUOB390xwgK/+2mrBMKs2Ta2Iy986Y0MOK96k=</SignatureValue><KeyInfo><X509Data><X509Certificate>
MIICczCCAdygAwIBAgIEAPymiTANBgkqhkiG9w0BAQUFADBrMQswCQYDVQQGEwJV
UzEQMA4GA1UECBMHTG90aGlhbjESMBAGA1UEBxMJRWRpbmJ1cmdoMQ0wCwYDVQQK
EwRUZXN0MRgwFgYJKoZIhvcNAQkBFglmcmVkQGhvbWUxDTALBgNVBAMTBGZyZWQw
HhcNMTAwMTA4MTczNzU4WhcNMTIwMTA4MTczNzU4WjBrMQswCQYDVQQGEwJVUzEQ
MA4GA1UECBMHTG90aGlhbjESMBAGA1UEBxMJRWRpbmJ1cmdoMQ0wCwYDVQQKEwRU
ZXN0MRgwFgYJKoZIhvcNAQkBFglmcmVkQGhvbWUxDTALBgNVBAMTBGZyZWQwgZ8w
DQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBALFSdWsBxphxHYTLHM1ETlhHItfK+xsj
mcOZ9YOJHXpsONyzNyxWJZCZSK2dmwGaxqvJEsxGo/vzKvwO5K9sAT5b5xlCJmbu
JA0XKTlLC0YTi+bNLK6db0RDoIfVfMhbW4k1Gpc/vmKWDdRAGR8phwmBQc0CqX+4
K/Vj/CCMD8hpAgMBAAGjJDAiMCAGCWCGSAGG+EIBDAQTFhF3d3cuYWJ5bG9uc29m
dC5kZTANBgkqhkiG9w0BAQUFAAOBgQA4YKdsSWLah3e0Ccpdzsqrjo/QxOvlEyRy
M1n8/Dszfxik1LurcFcM8tTuyKT9aQ2EnV/id8mKx01ldzFAPkWQWU1tappOfZ/2
bHOM8gdmiyHpKCrkkvVR+YJc0x3FuzjuqJP8CkC/WdTJr3YcuLQ6K9QhPO8KIVDs
3BvXsf4F9Q==

</X509Certificate><X509IssuerSerial><X509IssuerName>OU=,O=Test,C=US</X509IssuerName><X509SerialNumber>00fca689</X509SerialNumber></X509IssuerSerial></X509Data></KeyInfo></Signature></tns:RacunZahtjev>
  </soapenv:Body>
</soapenv:Envelope>"
`;
